### PR TITLE
Added new pre-checks to NUC to ensure Intel Turbo Boost is enabled

### PR DESCRIPTION
**Pull request by BenStarmerSmith**
## Proposed changes

Added new pre-checks to NUC to ensure Intel Turbo Boost is enabled. Fails with an error message if it isn't.

## Checklist

Before posting a PR ensure that from each of the below categories **AT LEAST ONE BOX HAS BEEN CHECKED**. If more than one category is applicable then more can be checked. Also ensure that the proposed changes have been filled out with relevant information for reviewers.

## Tests

- [ ] No tests required to be added. (For small changes that will be tested by CI/CD infrastructure).
- [ ] Added/Modified automated and PhantomHand CI tests (if a new class is added (Python or C++), the interface of that class must be unit tested).
- [ ] Manually tested in simulation (if simulation specific or no hardware required to test the functionality). 
- [x] Manually tested on hardware (if hardware specific or related).

## Documentation

- [x] No documentation required to be added.
- [ ] Added documentation (For any new feature, explain what it does and how to use it. Write the documentation in a relevant space, e.g. Github, Confluence, etc).
- [ ] Updated documentation (For changes to pre-existing features mentioned in the documentation).

### CODE REVIEW

**Comment by BenStarmerSmith** [comment]
Screenshot from 2023-05-09 15-41-23

**Comment by BenStarmerSmith** [comment]
Screenshot from 2023-05-09 16-01-51

**Review by jeferal — APPROVED** [review]

File: ansible/roles/installation/nuc-pre-check/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+---
+- name: Check if Intel Turbo Boost is enabled
+  shell: cat /sys/devices/system/cpu/intel_pstate/no_turbo
```
Comment (devops-shadow) [inline ansible/roles/installation/nuc-pre-check/tasks/main.yml:17]: Is this file always available?
Have you checked if you can use an inbuild function like this? https://docs.ansible.com/ansible/latest/collections/ansible/builtin/file_lookup.html

Comment (devops-shadow) [reply ansible/roles/installation/nuc-pre-check/tasks/main.yml:17]: Try to use ansible calls whenever possible. Use shell and cmd only when there isnt one available in ansible

Comment (BenStarmerSmith) [reply ansible/roles/installation/nuc-pre-check/tasks/main.yml]: Applied you suggestions :)

**Review by beatrizleon — APPROVED** [review]

**Pull request merged**